pkg/route/routeapihelpers/routeapihelpers.go

```diff
@@ -42,3 +44,29 @@ func IngressURI(route *routev1.Route, host string) (*url.URL, *routev1.RouteIngr
 	}
 	return nil, nil, fmt.Errorf("no ingress for host %s in route %s in namespace %s", host, route.ObjectMeta.Name, route.ObjectMeta.Namespace)
 }
+
+// ValidateHost checks that a route's host name satisfies DNS requirements.
+// Unless the allowNonCompliant annotation is set to true, host name must have
+// at least two labels, with each label less than 63 chars from a given character
+// set, and total length less than 253 chars.  Trailing dots are allowed.
+// If allowNonCompliant is set to true, only check that the host satisfies
+// basic DNS subdomain conditions, e.g. length and character set.
+func ValidateHost(host string, allowNonCompliant string) field.ErrorList {
+	result := field.ErrorList{}
+	specPath := field.NewPath("spec")
```

Comment:
It seems strange to construct the `field.Path` value and return a `field.ErrorList` value.  My inclination would be to make `ValidateHost` return a more general `error` value and leave it up to the caller to build the more specific `field.ErrorList` value.  However, `kvalidation.IsFullyQualifiedDomainName` needs a `field.Path` value, so maybe it would make more sense for `ValidateHost` both to have a `field.Path` parameter and a `field.ErrorList` return value like other validation functions.

Reply:
Sounds reasonable.